core/src/test/java/org/ehcache/EhcacheBasicGetAllTest.java

```diff
@@ -366,6 +443,99 @@ public void testGetAllStoreNoMatchCacheAccessExceptionBeforeLoaderAllFail() thro
     validateStats(ehcache, EnumSet.noneOf(CacheOperationOutcomes.GetOutcome.class));
     validateStats(ehcache, EnumSet.noneOf(CacheOperationOutcomes.CacheLoadingOutcome.class));
   }
+  
+  /**
+   * Tests {@link Ehcache#getAll(Set)} for
+   * <ul>
+   *    <li>non-empty request key set</li>
+   *    <li>no {@link Store} entries match</li>
+   *    <li>{@link Store#bulkComputeIfAbsent} throws before accessing loader</li>
+   *    <li>with a {@code CacheLoaderWriter} (loader-provided entries not relevant)</li>
+   *    <li>all {@link CacheLoaderWriter#loadAll(Iterable)} calls fail with {@link BulkCacheLoadingException}</li>
+   * </ul>
+   */
+  @Test
+  public void testGetAllStoreNoMatchCacheAccessExceptionBeforeLoaderAllFailWithBulkCacheLoadingException() throws Exception {
+    final FakeStore fakeStore = new FakeStore(getEntryMap(KEY_SET_B));
+    this.store = spy(fakeStore);
+    doThrow(new CacheAccessException("")).when(this.store)
+        .bulkComputeIfAbsent(getAnyStringSet(), getAnyIterableFunction());
+
+    final FakeCacheLoaderWriter fakeLoader = new FakeCacheLoaderWriter(TEST_ENTRIES, true);
+    this.loaderWriter = spy(fakeLoader);
+
+    final Ehcache<String, String> ehcache = this.getEhcache(this.loaderWriter);
+
+    try {
+      ehcache.getAll(KEY_SET_A);
+      fail();
+    } catch (BulkCacheLoadingException e) {
+      // Expected
+      assertThat(e.getSuccesses().keySet(), empty());
+      assertThat(e.getFailures().keySet(), Matchers.<Set<?>>equalTo(KEY_SET_A));
+    }
+
+    final InOrder ordered = inOrder(this.loaderWriter, this.spiedResilienceStrategy);
+    verify(this.store).bulkComputeIfAbsent(eq(KEY_SET_A), getAnyIterableFunction());
+    // ResilienceStrategy invoked: no assertion for Store content
+    ordered.verify(this.loaderWriter, atLeast(1)).loadAll(this.loadAllCaptor.capture());
+    assertThat(this.getLoadAllArgs(), equalTo(KEY_SET_A));
+
+    ordered.verify(this.spiedResilienceStrategy)
+        .getAllFailure(eq(KEY_SET_A), any(CacheAccessException.class), this.bulkExceptionCaptor.capture());
+    assertThat(this.bulkExceptionCaptor.getValue().getSuccesses().keySet(), empty());
+    assertThat(this.bulkExceptionCaptor.getValue().getFailures().keySet(), Matchers.<Set<?>>equalTo(KEY_SET_A));
```

Comment:
You have some duplication caused by the fact that you check here the exception but also in the catch block above.
I would either check here that we have the same exception or extract that code to an exception check method:
`checkBulkLoadingException(BulkCacheLoadingException e, Set<K> successKeys, Set<K> failureKeys)`